src/lib.rs

```diff
@@ -30,6 +30,31 @@ mod unix;
 #[cfg(unix)]
 use crate::unix as sys;
 
+#[cfg(all(unix, feature = "multilock"))]
+mod unix_fileid;
+#[cfg(all(unix, feature = "multilock"))]
+use unix_fileid as fileid;
+#[cfg(not(all(unix, feature = "multilock")))]
+mod nil_fileid;
+#[cfg(not(all(unix, feature = "multilock")))]
+use nil_fileid as fileid;
+
+/// Enumeration used to declare whether FsLock instances opened with the same
+/// file, by the same process, are exclusive.
+#[derive(Debug,Copy,Clone)]
+#[non_exhaustive]
+enum Exclusive {
+    /// Treat any two file descriptors to the same file as having
+    /// separate locks.
+    ///
+    /// This option requires allocation internally, and is not
+    /// available on Unix when building without the `std` feature.
+    #[cfg(any(not(unix), feature="multilock"))]
+    ExclusiveInProcess,
```

Comment:
Since this type now encodes manners of being exclusive, this variant could be renamed to "per file descriptor".
```suggestion
    PerFileDesc,
```